brightwind/transform/transform.py

```diff
@@ -1400,3 +1401,276 @@ def offset_timestamps(data, offset, date_from=None, date_to=None, overwrite=Fals
+                                            DATE_INSTEAD_OF_NONE else meas_station_data_model_from)
+                meas_station_data_model_to = meas_station_data_model.get('date_to')
+                
+                if date_to is None or date_to == DATE_INSTEAD_OF_NONE:
+                    date_to_tmp = meas_station_data_model_to
+                else:
+                    idx_pos = df.index.get_indexer([pd.Timestamp(date_to)], method='nearest')[0]
+                    date_to_tmp = df.index[idx_pos + 1].strftime('%Y-%m-%dT%H:%M:%S') if idx_pos + 1 < len(df.index) else date_to
+
+                if i==0 or previous_name != name:
+                    previous_name = name
+                    i+=1
+                if (((meas_station_data_model_to is None) or (date_from <= meas_station_data_model_to)) and 
+                    ((date_to_tmp is None) or (meas_station_data_model_from is None) or 
+                     (date_to_tmp >= meas_station_data_model_from))):
+                    device_orientation_deg = meas_station_data_model.get('device_orientation_deg')
+                    apply_offset_from = (date_from if date_from > meas_station_data_model_from 
+                                         else meas_station_data_model_from)
+                    if date_to_tmp is None or meas_station_data_model_to is None:
+                        apply_offset_to = date_to_tmp if date_to_tmp is not None else meas_station_data_model_to
+                    else:
+                        apply_offset_to = min(date_to_tmp, meas_station_data_model_to)
+
+                    previous_name = name
+                    df[name] = _apply_dir_offset_target_orientation(
+                        df[name], logger_offset, device_orientation_deg, apply_offset_from, apply_offset_to,
+                        target_orientation_name='device orientation')
+        else:
+            wdir_not_in_dataset = True
+            col_not_in_data.append(name)
+    
+    if wdir_not_in_dataset:
+        indexes = np.unique(col_not_in_data, return_index=True)[1]
+        col_not_in_data = [col_not_in_data[index] for index in sorted(indexes)]
+        print_text = 'Following wind direction measurement(s) not found in the data'
+        if wdir_cols:
+            print(print_text + ' for the requested `wdir_cols`: {}.'.format(utils.bold(str(col_not_in_data))))
+        else:
+            print(print_text + ': {}.'.format(utils.bold(str(col_not_in_data))))
+    if col_not_in_datamodel:
+        print('No device orientation offset applied to following requested measurement(s) as no wind direction measurement '
+        'type found in `meas_station_data_models` for these: {}.'.format(utils.bold(str(col_not_in_datamodel))))
+    # if a Series is sent, send back a Series
+    if isinstance(data, pd.Series):
+        df = df[df.columns[0]]
+        data.update(df)
+    return df
+
+
+def _check_vertical_profiler_properties_not_overlap(meas_station_data_models, df):
```

Comment:
Doesn't look like a great name with the negative near the end. Could it not be a True/False question so `_check_vertical_profiler_properties_overlap()`? See below comment too.